src/ref_serializer.rs

```diff
@@ -66,60 +77,59 @@ impl<'a> RefParameterSerializer<'a> {
 ///     .unwrap()
 ///     .close_inner_list()
 ///     .parameter("bar", RefBareItem::String("val"))
+///     .unwrap()
+///     .finish()
 ///     .unwrap();
 /// assert_eq!(
-///     serialized_item,
+///     serialized_list,
 ///     "11;foo, (abc;abc_param=?0 def);bar=\"val\""
 /// );
 /// ```
 #[derive(Debug)]
-pub struct RefListSerializer<'a> {
-    buffer: &'a mut String,
+pub struct RefListSerializer {
+    buffer: String,
 }
 
-impl<'a> RefListSerializer<'a> {
-    pub fn new(buffer: &'a mut String) -> Self {
-        RefListSerializer { buffer }
+impl RefListSerializer {
+    pub fn new() -> Self {
+        RefListSerializer {
+            buffer: String::new(),
+        }
     }
 
-    pub fn bare_item(self, bare_item: impl AsRefBareItem) -> SFVResult<Self> {
-        if !self.buffer.is_empty() {
-            self.buffer.push_str(", ");
-        }
-        Serializer::serialize_bare_item(bare_item, self.buffer)?;
-        Ok(RefListSerializer {
-            buffer: self.buffer,
-        })
+    pub fn bare_item(mut self, bare_item: impl AsRefBareItem) -> SFVResult<Self> {
+        write_separator(&mut self.buffer);
+        Serializer::serialize_bare_item(bare_item, &mut self.buffer)?;
+        Ok(self)
     }
 
-    pub fn parameter(self, name: &str, value: impl AsRefBareItem) -> SFVResult<Self> {
+    pub fn parameter(mut self, name: &str, value: impl AsRefBareItem) -> SFVResult<Self> {
         if self.buffer.is_empty() {
             return Err("parameters must be serialized after bare item or inner list");
         }
-        Serializer::serialize_parameter(name, value, self.buffer)?;
-        Ok(RefListSerializer {
-            buffer: self.buffer,
-        })
+        Serializer::serialize_parameter(name, value, &mut self.buffer)?;
+        Ok(self)
     }
-    pub fn open_inner_list(self) -> RefInnerListSerializer<'a, Self> {
-        if !self.buffer.is_empty() {
-            self.buffer.push_str(", ");
-        }
+
+    pub fn open_inner_list(mut self) -> RefInnerListSerializer<Self> {
+        write_separator(&mut self.buffer);
         self.buffer.push('(');
-        RefInnerListSerializer::<RefListSerializer> {
-            buffer: self.buffer,
-            caller_type: PhantomData,
+        RefInnerListSerializer { buffer: self }
+    }
+
+    pub fn finish(self) -> SFVResult<String> {
+        if self.buffer.is_empty() {
+            return Err("serializing empty list is not allowed");
```

Comment:
I think the RFC allows this?

Reply:
See https://github.com/undef1nd/sfv/issues/132#issuecomment-2671490077.